src/Fields/interpolate.jl

```diff
@@ -0,0 +1,55 @@
+# Converting (x, y, z) coordinates to indices on a RegularCartesianGrid.
+# The returned indices are floats indicating a location between grid points.
+@inline x_to_index(x, ::Type{Cell}, grid) = @inbounds (x - grid.xC[1]) / grid.Δx
```

Comment:
Ah you're right but unfortunately here the method would have to be different for a stretched grid because you have to iterate over the indices until you reach an index `k` such that `grid.z[k] < z < grid.z[k+1]`.

So I'll dispatch on `::RegularCartesianGrid` to indicate that this only works for regular Cartesian grids for now.

Reply:
Or just use `searchsortedfirst`.

Reply:
`searchsortedfirst` is almost 4x slower and doesn't return the fractional indices (only integer indices). You need the fractional indices to compute the unit cell indices. Perhaps there is a version that returns fractional indices, I'll look around.

```
julia> @benchmark x_to_index(5.0, Cell, $grid)
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     19.609 ns (0.00% GC)
  median time:      20.112 ns (0.00% GC)
  mean time:        21.986 ns (0.00% GC)
  maximum time:     5.753 μs (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     997

julia> @benchmark searchsortedfirst($grid.xC, 5.0)
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     72.738 ns (0.00% GC)
  median time:      76.374 ns (0.00% GC)
  mean time:        83.856 ns (0.00% GC)
  maximum time:     13.017 μs (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     971

```

Reply:
I meant for stretched grids.

Reply:
We could add an implementation for stretched grids that uses `searchsortedfirst` to compute the integer index then figures out the fractional index from that, but we'd have no way of testing it right now.

Reply:
> We could add an implementation for stretched grids that uses searchsortedfirst to compute the integer index then figures out the fractional index from that, but we'd have no way of testing it right now.

Yes I agree with that. I was responding to this comment:

> unfortunately here the method would have to be different for a stretched grid because you have to iterate over the indices until you reach an index k such that grid.z[k] < z < grid.z[k+1].

by pointing out that you could use `searchsortedfirst` instead.